src/common/middleware/call-api.js

```diff
@@ -0,0 +1,101 @@
+import { checkStatus, getError } from '../helpers/api';
+
+// Action key that carries API call info interpreted by this Redux middleware.
+export const CALL_API = 'CALL_API';
+
+// A Redux middleware that interprets actions with CALL_API info specified.
+// Performs the call and promises when such actions are dispatched.
+export default defaults => () => next => async action => {
+  // If action does not invoke CALL_API,
+  // ignore it
+  const settings = action[CALL_API];
+  if (typeof settings === 'undefined') {
+    return next(action);
+  }
+
+  // Validate action settings
+  validateDefaults(defaults);
+  validateSettings(settings);
+
+  const [ successType, failureType ] = settings.types;
+  const resource = defaults.endpoint + settings.path;
+  const optionsWithCsrfToken = withCsrfToken(defaults.csrfToken, settings.options);
+  const createAction = createActionWith.bind({}, action);
+
+  // If action has a type, dispatch the action next
+  if (action.type) {
+    next(createAction());
+  }
+
+  try {
+    const response = await fetch(resource, optionsWithCsrfToken);
+    await checkStatus(response);
+    const result = await response.json();
+
+    if (result.status !== 'success') {
+      throw getError(response, result);
+    }
+
+    return next(createAction({
+      type: successType,
+      response: result.payload
+    }));
+  }
+  catch(error) {
+    return next(createAction({
+      type: failureType,
+      error: error
```

Comment:
Yep, it's quite unfortunate that with such a freedom redux gives it doesn't really enforce any conventions, so every example use it in different way.

I thought it was nice for us to try to use FSA as much as we could as they give us this convention that redux doesn't on its own.
Unfortunately - the redux 'real world' example - doesn't use it, and if we want to follow its patters we have to leave FSA behind.

That being said, I'm not sure if we need to tread redux real-world example as an official standard documentation. It's just an example of possible implementation. Copying it one by one doesn't make it better then anything else. Doesn't make it more 'redux standard'.
So maybe it would be possible to make our CALL_API middleware compatible with FSA in some way?